docs/rfc/grouping-api.md

```diff
@@ -0,0 +1,53 @@
+# Grouping API
+
+We will be adding the ability to add 'grouping' support to `react-beautiful-dnd`. I am attempting to create naming for the new api that feels right. The grouping feature will enable ability to hover over a Draggable to group with it. Right now I have been calling the feature 'grouping'. However, I am not sure if even that is the right name for the feature.
+
+## API
+
+### `Draggable` > isGroupingEnabled
+
+This would allow a particular Draggable to be grouped with
+
+### `Draggable` > `DraggableStateSnapshot`
+
+```diff
+export type StateSnapshot = {|
+  isDragging: boolean,
+  draggingOver: ?DroppableId,
+-  isDropAnimating: boolean,
++  dropping: ?DroppingState,
++  groupingWith: ?DraggableId,
++  groupedOverBy: ?DraggableId,
+|};
+
++export type DroppingState = {|
```

Comment:
`reason` and `duration` definitely make sense to me, I assume it's `DroppingState` that feels weird?

Not sure on a great name either. `DroppingMeta`, `DroppingData`... I'll think on it